src/LRS.js

```diff
@@ -454,27 +455,35 @@ TinCan client library
             }
             if (typeof cfg.callback !== "undefined") {
                 callbackWrapper = function (err, xhr) {
-                    var result = xhr.responseText,
+                    var result,
                         boundary,
                         section,
+                        headers,
                         statement,
+                        attachmentMap = {},
                         i;
 
                     if (err === null) {
                         if (! cfg.params.attachments) {
-                            result = TinCan.Statement.fromJSON(result);
+                            result = TinCan.Statement.fromJSON(xhr.responseText);
                         }
                         else {
                             boundary = xhr.getResponseHeader("Content-Type").split("boundary=")[1];
-                            result = result.split("--" + boundary + "--")[0].split("--" + boundary);
+                            result = xhr.responseText.split("--" + boundary + "--")[0].split("--" + boundary);
 
                             statement = JSON.parse(result[1].split("\r\n\r\n")[1]);
 
                             for (i = 2; i < result.length; i += 1) {
                                 section = result[i].split("\r\n\r\n");
-                                statement = TinCan.LRS.prototype._assignAttachmentContent({statements: [statement]}, section[0].split("\r\n"), section[1].trim()).statements[0];
+                                headers = section[0].split("\r\n");
```

Comment:
I'm not convinced any of these `split`s will actually work as intended. They'll handle the simple case correctly, but will fall down on either complex headers, and/or bodies that include non-normal whitespace. Minimally they need to include the second limit argument, specifying that the split should only occur once. For instance, headers can be split across lines or bodies might include multiple line breaks in a row, hence the separation of header from body + boundary.